apps/nextjs/pages/login/[provider].tsx

```diff
@@ -0,0 +1,88 @@
+import { GetServerSideProps } from "next";
+import dynamic from "next/dynamic";
+import Router from "next/router";
+import { useEffect } from "react";
+
+import { ContentWrapper } from "components/Wrapper";
+import { AuthCompleteResponse } from "pages/api/auth/complete";
+import { AUTH_PROVIDERS } from "utils/constants";
+import { getJsonPostSettings } from "utils/fetch";
+import { getString } from "utils/urlQuery";
+
+interface PageProps {
+  code: string | null;
+  provider: string;
+  state: string | null;
+}
+
+function loginPage(props: PageProps): JSX.Element | null {
+  const { code, provider, state: receivedState } = props;
+  const expectedState = localStorage.getItem(`${provider}LoginState`);
+
+  if (expectedState === null) {
+    throw new Error(
+      `AuthError: state token for ${provider} not found in localStorage`
+    );
+  }
+
+  if (receivedState !== expectedState) {
+    throw new Error(
+      `AuthError: state token received from ${provider} doesn't match expected value`
+    );
+  }
+
+  if (code === null) {
+    throw new Error('AuthError: missing required parameter "code"');
+  }
+
+  useEffect(() => {
+    (async () => {
+      const res = await postNextBackend(provider, code);
+      localStorage.removeItem(`${provider}LoginState`);
+
+      if (!res.ok) {
+        throw new Error(`AuthError: ${res.error}`);
+      }
+
+      // TODO: store res.sessionId to global state.
+      Router.push("/");
+    })();
+  }, []);
```

Comment:
should depend on `provider` & `code` right? That said, I'd look into making the auth completion query a standalone hook as that'll help decoupling the logic from the templating (and is easier to convert to use suspenses if/when needed).

Reply:
AFAIK using empty `deps` is ok when you want to run the effect only once. And since this page's purpose in life is to receive a redirect from OAuth provider, check the `state` and relay the `code` to backend(s), I don't expect `provider` or `code` to change.

I'll look into making this a custom hook.

Reply:
That's true (you can keep it empty if you want only-once semantics), though it'd still be best practice to use the deps (if the deps don't change, the hook is still run only once after all). In practice it probably doesn't matter here at all, but I know some linters will complain if you don't declare the deps you are using within the effect.

Reply:
dded a [custom hook](https://github.com/thunderstore-io/thunderstore-ui/pull/404/commits/0498117426766569ce94a51cd56871c969aeac43#diff-6b5355ca43f78707309fbc67299be7f0aa2e37d9ad12885593aae843e344f9d7)

As you can see, the linters are not complaining here. I feel that the "we should change our lint settings" are discussions that should take place outside feature-related PRs. Left the deps as was.

Reply:
Well, in this case, if `code` does change after the initial page call, it's probably a sign that someone's trying to brute force the login, and not sending another request to backend might be what we want. But I guess that should be handled on backend's side anyway.

To avoid further time spent on this discussion, I added the `deps`.